build-tools/packages/build-cli/src/commands/report/codeCoverage.ts

```diff
@@ -0,0 +1,166 @@
+			env: "ADO_CI_BUILD_DEFINITION_ID_BASELINE",
+			required: true,
+		}),
+		ADO_CI_BUILD_DEFINITION_ID_PR: Flags.integer({
+			description: "Build definition/pipeline number/id for the PR build.",
+			env: "ADO_CI_BUILD_DEFINITION_ID_PR",
+			required: true,
+		}),
+		CODE_COVERAGE_ANALYSIS_ARTIFACT_NAME_BASELINE: Flags.string({
+			description: "Code coverage artifact name for the baseline build.",
+			env: "CODE_COVERAGE_ANALYSIS_ARTIFACT_NAME_BASELINE",
+			required: true,
+		}),
+		CODE_COVERAGE_ANALYSIS_ARTIFACT_NAME_PR: Flags.string({
+			description: "Code coverage artifact name for the PR build.",
+			env: "CODE_COVERAGE_ANALYSIS_ARTIFACT_NAME_PR",
+			required: true,
+		}),
+		GITHUB_PR_NUMBER: Flags.integer({
+			description: "Github PR number.",
+			env: "GITHUB_PR_NUMBER",
+			required: true,
+		}),
+		GITHUB_REPOSITORY_NAME: Flags.string({
+			description: "Github repository name.",
+			env: "GITHUB_REPOSITORY_NAME",
+			required: true,
+		}),
+		GITHUB_REPOSITORY_OWNER: Flags.string({
+			description: "Github repository owner.",
+			env: "GITHUB_REPOSITORY_OWNER",
+			required: true,
+		}),
+		TARGET_BRANCH_NAME: Flags.string({
+			description: "Target branch name.",
+			env: "TARGET_BRANCH_NAME",
+			required: true,
+		}),
+		...BaseCommand.flags,
+	};
+
+	public async run(): Promise<void> {
+		const { flags } = this;
+
+		const codeCoverageConstantsForBaseline: IAzureDevopsBuildCoverageConstants = {
+			orgUrl: "https://dev.azure.com/fluidframework",
+			projectName: "public",
+			ciBuildDefinitionId: flags.ADO_CI_BUILD_DEFINITION_ID_BASELINE,
+			artifactName: flags.CODE_COVERAGE_ANALYSIS_ARTIFACT_NAME_BASELINE,
+			branch: flags.TARGET_BRANCH_NAME,
+			buildsToSearch: 50,
+		};
+
+		const codeCoverageConstantsForPR: IAzureDevopsBuildCoverageConstants = {
+			orgUrl: "https://dev.azure.com/fluidframework",
+			projectName: "public",
+			ciBuildDefinitionId: flags.ADO_CI_BUILD_DEFINITION_ID_PR,
+			artifactName: flags.CODE_COVERAGE_ANALYSIS_ARTIFACT_NAME_PR,
+			buildsToSearch: 20,
+			buildId: flags.ADO_BUILD_ID,
+		};
+
+		// Get the paths of the files that have changed in the PR relative to root of the repo.
+		// This is used to determine which packages have been affect so that we can do code coverage
+		// analysis on those packages only.
+		const changedFiles = await getChangedFilePaths(
+			{
+				owner: flags.GITHUB_REPOSITORY_OWNER,
+				repo: flags.GITHUB_REPOSITORY_NAME,
+				token: flags.GITHUB_API_TOKEN,
+			},
+			flags.GITHUB_PR_NUMBER,
+		);
+
+		let commentMessage: string = "";
+		const report = await getCodeCoverageReport(
+			flags.ADO_API_TOKEN,
+			codeCoverageConstantsForBaseline,
+			codeCoverageConstantsForPR,
+			changedFiles,
+			this.logger,
+		).catch((error: Error) => {
+			commentMessage = "## Code Coverage Summary\n\nError getting code coverage report";
+			this.logger.errorLog(`Error getting code coverage report: ${error}`);
+			return undefined;
+		});
```

Comment:
Is there an advantage to using .catch rather than wrapping in a try/catch?

Reply:
Since we are assigning to report in catch case too, it is good in terms of code structure rather than a try/catch.